### ISSUE TITLE
Simple Sidearms - automatic weapon swap doesn't happen

### ISSUE THREAD

**Issue by endlesik** [NoImageAvailable/CombatExtended#306]
Simple Sidearms works as long as CE isn't enabled - when it is enabled however - pawns will carry multiple weapons etc, but they won't automatically swap those. Tried using only:

1. Core
2. JecsTools
3. Huggslib
4. CE
5. SS  - tried to swap SS with CE - but it still didn't work.

Tried with and without ammo system from CE, and multiple settings from SS, still coulnd't get it to work.
SS v1.2.1 and CE 1.4c

EDIT

I've talked to a friend that has CE and SS working - we have the same version of Rimworld 17b v0.17.1557 -
 I set everything as he did - mods order, specific mods settings etc., and he even send me his mods files, which I tried to use - still, pawns won't swap weapons etc. Ah, and the console don't return any errors about SS or CE (or at least I don't see any).
Below are two game captures - on the first one CE is disabled and everything works like a charm. On second capture, with CE enabled... well, SS automatic swap doesn't work at all.

![rim1](https://user-images.githubusercontent.com/30985116/29296213-2a31205c-8159-11e7-8e5c-5500a964c41c.gif)
![rim3](https://user-images.githubusercontent.com/30985116/29296275-8c367568-8159-11e7-9b65-57b3773f1afb.gif)

About reproducing - all I did was to create a new world (for each test), gave weapons to colonist and spawned an enemy, and let them do it's things.

**Comment by krudowicz** [comment]
As I've been mentioned here, my instance works partially.
Order of CE/SS:
- Combat Extended Core
- Combat Extended Guns
- SimpleSidearms

Above those two are several mods/tools, including JecsTools and HugsLib.

The weird thing is, for three separate tries, that I did to present those two mods working together, first succeded, the later two failed. As for this moment, I cannot reproduce the working behaviour anymore and unfortunately I was not recording on the first try.

The scenario, when it worked fine was with autoswitch on and "Fire free" disabled to not kill poor tribal man before he joins into melee.
Weapons consisted of AK with several magazines and an awful uranium knife. The knife was equipped as a sidearm, while the pawn was set to be drafted. It also may have impacted the result, but others try performed in the same way did actually fail.

For me it does not depend on any amount of ammo or lack of it, the pawn is properly dropping his ranged weapon and switching to meele weapon(knife in my instance).
I had set the SS settings to advanced preset and, IIRC, nothing else was changed in the menu.

I'm not able to provide video of working case for now. If I will aquire a viable, helpful fragment, I will add it.
The issue may be also dependent on other mods, as we have a little bit different stacks of them and our order does not exactly match in every way.
